PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,191 @@
+            Logger.debug(isAppBackgrounded
+                            ? Strings.purchaserInfo.purchaserinfo_stale_updating_in_background
+                            : Strings.purchaserInfo.purchaserinfo_stale_updating_in_foreground)
+            fetchAndCachePurchaserInfo(appUserID: appUserID,
+                                       isAppBackgrounded: isAppBackgrounded,
+                                       completion: completion)
+            Logger.rcSuccess(Strings.purchaserInfo.purchaserinfo_updated_from_network)
+        } else {
+            if let completion = completion {
+                operationDispatcher.dispatchOnMainThread {
+                    completion(cachedPurchaserInfo, nil)
+                }
+            }
+        }
+    }
+
+    @objc(sendCachedPurchaserInfoIfAvailableForAppUserID:)
+    public func sendCachedPurchaserInfoIfAvailable(appUserID: String) {
+        guard let info = cachedPurchaserInfo(appUserID: appUserID) else {
+            return
+        }
+
+        sendUpdateIfChanged(purchaserInfo: info)
+    }
+
+    @objc(purchaserInfoWithAppUserID:completionBlock:)
+    public func purchaserInfo(appUserID: String, completionBlock: ReceivePurchaserInfoBlock?) {
+        let infoFromCache = cachedPurchaserInfo(appUserID: appUserID)
+        var completionCalled = false
+
+        if let infoFromCache = infoFromCache {
+            Logger.debug(Strings.purchaserInfo.vending_cache)
+            if let completion = completionBlock {
+                completionCalled = true
+                operationDispatcher.dispatchOnMainThread {
+                    completion(infoFromCache, nil)
+                }
+            }
+        }
+
+        // Prevent calling completion twice.
+        let fetchPurchaserInfoCompletion = completionCalled ? nil : completionBlock
+
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            self.fetchAndCachePurchaserInfoIfStale(appUserID: appUserID,
+                                                   isAppBackgrounded: isAppBackgrounded,
+                                                   completion: fetchPurchaserInfoCompletion)
+        }
+    }
+
+    @objc(cachedPurchaserInfoForAppUserID:)
+    public func cachedPurchaserInfo(appUserID: String) -> PurchaserInfo? {
+        guard let purchaserInfoData = deviceCache.cachedPurchaserInfoData(appUserID: appUserID) else {
+            return nil
+        }
+
+        do {
+            let maybeInfoDict = try JSONSerialization.jsonObject(with: purchaserInfoData) as? [String: Any]
+            guard let purchaserInfoDict = maybeInfoDict, let info = PurchaserInfo(data: purchaserInfoDict) else {
+                return nil
+            }
+
+            if let schema = info.schemaVersion, schema == PurchaserInfo.currentSchemaVersion {
+                return info
+            }
+        } catch {
+            Logger.warn("Unable to unmarshall PurchaserInfo from cache:\n \(error.localizedDescription)")
```

Comment:
I think this merits `Logger.error`, this is a situation where the SDK can't update the purchaserInfo

Reply:
Cool, I wasn't 100% sure when we do warn vs error

Reply:
It's all good 😄  logging is hard.